### PR TITLE
Ensure correct bin_dir for pg params configuration

### PR DESCRIPTION
**Pull request by hughcapet**
Make inplace upgrade code use the bin_dir of the proper version throught the whole process, so that
ConfigHandler.write_postgresql_conf() validates provided params using the proper PG version.

### CODE REVIEW

File: postgres-appliance/tests/docker-compose.yml [hunk 1]
```diff
@@ -46,6 +46,7 @@ services:
                   postgresql:
                     parameters:
                       wal_keep_segments: 8
+                      jit: 'off'
```
Comment (macedigital) [inline postgres-appliance/tests/docker-compose.yml:49]: This is unreleated to the `bin_dir` changes, correct?

Comment (hughcapet) [reply postgres-appliance/tests/docker-compose.yml:49]: it is the parameter that exists in PG11+ and causes 10->N upgrade failure if bin_dir is used incorrectly
However given we are getting rid of 10 (and likely 11) soon, I should probably add other parameters for regression testing in the future.

**Comment by FxKu** [comment]
:+1:

File: postgres-appliance/tests/docker-compose.yml (hunk 1 above)
Comment (macedigital) [reply postgres-appliance/tests/docker-compose.yml:49]: Understood. Let's keep the parameter, especially given that PG10 is end-of-life by now and not be used much longer.

**Review by macedigital — COMMENTED** [review]
It took me a couple of passes to get an understanding of the changes, because the broader context of the full code base must be considered.

This the pull-request _should_ fix the failed upgrade from 10 -> 15 which triggered this PR, but I'm honestly not 100% certain all side-effects are caught.

**Comment by macedigital** [comment]
:+1:

**Comment by macedigital** [comment]
:+1:

**Comment by hughcapet** [comment]
👍

**Pull request merged**